src/consensus/aft/raft.h

```diff
@@ -1200,7 +908,11 @@ namespace aft
           }
         }
       }
-      else if (consensus_type != ConsensusType::BFT)
+      else if (
+        consensus_type != ConsensusType::BFT &&
+        replica_state != kv::ReplicaState::Retired &&
+        replica_state != kv::ReplicaState::Retiring &&
+        replica_state != kv::ReplicaState::Learner)
```

Comment:
Is this a functional change for the CFT path?

Reply:
Huh, this looks like a bad merge, and is not deliberate. I'll take a look.

Reply:
Those conditions are unnecessary, because `can_endorse_primary` also checks them.

Reply:
Yeah this was a bad merge. Have restored this check to the `else if (consensus_type != ConsensusType::BFT)` that is on `main`.